module/Core/src/Core/Controller/Plugin/CreatePaginator.php

```diff
@@ -82,8 +86,26 @@ public function __invoke($paginatorName, $defaultParams = array(), $usePostParam
             }
         }
 
+        /* try to create $paginator from event listener */
+        /* @var ResponseCollection $response */
+        /* @var EventManager $eventManager */
         /* @var $paginator \Zend\Paginator\Paginator */
-        $paginator = $paginators->get($paginatorName, $params);
+        $eventManager = $this->serviceManager->get('EventManager');
+        $response      = $eventManager->trigger(
+            static::EVENT_CREATE_PAGINATOR,
+            $this,
+            [
+                'params' => $params,
+                'paginatorName' =>$paginatorName,
+            ]
+        );
+        $paginator = $response->last(); // get the last result from listener
+
```

Comment:
This where I trigger event triggers ```core.paginator``` event as mentioned by @TiSiE in #234 issue.

CC:
@TiSiE @cbleek